sdk/python/kfp/v2/dsl/container_op.py

```diff
@@ -215,3 +217,23 @@ def set_gpu_limit(self, count: int) -> 'ContainerOp':
                        '{}'.format(count))
     self.container_spec.resources.accelerator.count = count
     return self
+
+  # Override _set_metadata to use v2 ComponentSpec
+  def _set_metadata(self, metadata):
```

Comment:
Is it possible to do type hint here?
```suggestion
  def _set_metadata(self, metadata: structures.ComponentSpec):
```

Reply:
Done.